specs/PushNotifications/ToastNotifications-spec.md

```diff
@@ -0,0 +1,509 @@
+
+```cpp
+//<toast launch="action=openThread&amp;threadId=92187">
+//  <visual>
+//      <binding template="ToastGeneric">
+//          <text hint-maxLines="1">Jill Bender</text>
+//          <text>Check out where we camped last weekend! Incredible!!</text>
+//          <image placement="hero" src="https://unsplash.it/360/180?image=1043" />
+//      </binding>
+//  </visual>
+//  <actions>
+//      <input id="textBox" type="text" placeHolderContent="reply" / >
+//      <action
+//          content="Send"
+//          imageUri="Assets/Icons/send.png"
+//          hint-inputId="textBox"
+//          activationType="background"
+//          arguments="action=reply&amp;threadId=92187" />
+//  </actions>
+//</toast>
+
+void ProcessNotificationArgs(const winrt::AppNotificationActivatedEventArgs& notificationActivatedEventArgs)
+{
+    // If the user clicks on a toast, the code will need to launch the chat thread window
+    if (std::wstring(notificationActivatedEventArgs.ActivationArgs().c_str()).find(L"openThread") != std::wstring::npos)
+    {
+        GenerateChatThreadWindow();
+    }
+    else // If the user responds to a message by clicking a button in the notification, we will need to reply back to the other user
+    if (std::wstring(notificationActivatedEventArgs.ActivationArgs().c_str()).find(L"reply") != std::wstring::npos)
+    {
+        auto input = notificationActivatedEventArgs.UserInput();
+        auto replyBoxText = input.Lookup(L"ReplyBox");
+
+        // Process the reply text
+        SendReplyToUser(replyBoxText);
+    }
+}
+```
+
+## Processing Active Notifications
+
+We define Active Notifications as payloads that have already been displayed by the app and show up
+in the Action Centre. In this example, the developer want to delete all Active Notifications from a
+former friend except if the Notification falls under the Archived group.
+
+```cpp
+winrt::Windows::Foundation::IAsyncAction RemoveAllNotificationsFromFormerFriend(std::wstring const& friendId)
+{
+    winrt::AppNotificationManager manager = winrt::AppNotificationManager::Default();
+    auto notifications = co_await manager.GetAllAsync();
+
+    for (auto notification : notifications)
+    {
+        // We remove all notifications with the friend who sent them.The tag is the friendId.
+        // We target all possible chat groups where this friend is present except the Archived group
+        if (notification.Tag() == friendId && notification.Group() != L"Archived")
+        {
+            co_await manager.RemoveByTagAndGroupAsync(notification.Tag(), notification.Group());
+        }
+    }
+}
+```
+
+## Notification Progress Updates
+
+Sometimes a developer would like to show progress bar related updates in a Notification:
+
+![Notification Progress](https://docs.microsoft.com/windows/apps/design/shell/tiles-and-notifications/images/toast-progressbar-annotated.png)
+
+To accomplish that, the developer will need to use the AppNotificationProgressData construct to
```

Comment:
"To accomplish that, use `AppNotificationProgressData` to describe updates to the progress UI presented in the notification"